### PR TITLE
Validate demisto in docs and description

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/25118

## Description
Added a validation for the word demisto in README and description.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -154,6 +155,7 @@
     "description_contains_contrib_details": {'code': "DS105", 'ui_applicable': False,
                                              'related_field': 'detaileddescription'},
     "invalid_description_name": {'code': "DS106", 'ui_applicable': False, 'related_field': ''},
+    "description_contains_demisto_word": {'code': "DS107", 'ui_applicable': False, 'related_field': ''},
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:158]:
```suggestion
    "description_contains_demisto_word": {'code': "DS107", 'ui_applicable': True, 'related_field': 'detaileddescription'},
```
You can fix this through the UI by taking the word out of the description

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -1294,6 +1301,11 @@ def readme_equal_description_error():
         return 'README.md content is equal to pack description. ' \
                'Please remove the duplicate description from README.md file.'
 
+    @staticmethod
+    @error_code_decorator
+    def readme_contains_demisto_word():
+        return 'Found the word \'Demisto\' in the readme content.'
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1307]: is there a way to specify a line?

File: demisto_sdk/commands/common/errors.py [hunk 3]
```diff
@@ -1019,6 +1021,11 @@ def invalid_description_name():
         return "The description's file name is invalid - " \
                "make sure the name looks like the following: <integration_name>_description.md"
 
+    @staticmethod
+    @error_code_decorator
+    def description_contains_demisto_word():
+        return 'Found the word \'Demisto\' in the description content.'
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1027]: is there a way to specify a line?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 * Fixed an issue where the **Assume yes** flag did not work properly for some entities in the **format** command.
+* Added a validation to ensure that description and README does not contains the word 'Demisto'.
```
Comment (roysagi) [inline CHANGELOG.md:3]:
```suggestion
* Added a validation to ensure that description and README do not contain the word 'Demisto'.
```

**Review by roysagi — COMMENTED** [review]
Looks good though if we can help the user by adding info as to where does the word Demisto was caught that might be good.

File: demisto_sdk/commands/common/errors.py (hunk 2 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/errors.py:1308]: done.

File: demisto_sdk/commands/common/hook_validations/description.py [hunk 5]
```diff
@@ -137,3 +138,44 @@ def is_valid_description_name(self):
...
+                if self.handle_error(error_message, error_code, file_path=self.file_path, warning=True):
+                    return True
+
+            with open(description_path) as f:
+                description_content = f.read()
+
+        for line_num, line in enumerate(description_content.split('\n')):
+            if 'demisto ' in line.lower() or ' demisto' in line.lower():
+                error_message, error_code = Errors.description_contains_demisto_word(line_num + yml_line_num + 1)
+                self.handle_error(error_message, error_code, file_path=self.file_path)
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/description.py:177]:
```suggestion
                if self.handle_error(error_message, error_code, file_path=self.file_path)
```

File: demisto_sdk/commands/common/hook_validations/description.py [hunk 6]
```diff
@@ -137,3 +138,44 @@ def is_valid_description_name(self):
...
+            except IndexError:
+                error_message, error_code = Errors.no_description_file_warning()
+                if self.handle_error(error_message, error_code, file_path=self.file_path, warning=True):
+                    return True
+
+            with open(description_path) as f:
+                description_content = f.read()
+
+        for line_num, line in enumerate(description_content.split('\n')):
+            if 'demisto ' in line.lower() or ' demisto' in line.lower():
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/description.py:175]: let's gather all the lines which have the word in them and give one error back - not multiple times.

File: demisto_sdk/commands/common/hook_validations/readme.py [hunk 7]
```diff
@@ -357,6 +357,24 @@ def is_context_different_in_yml(self) -> bool:
...
+        Checks if there are the word 'Demisto' in the README content.
+
+        Return:
+            True if 'Demisto' does not exist in the README content, and False if it does.
+        """
+
+        is_valid = True
+
+        for line_num, line in enumerate(self.readme_content.split('\n')):
+            if 'demisto ' in line.lower() or ' demisto' in line.lower():
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/readme.py:371]: let's gather all the lines which have the word in them and give one error back - not multiple times.

File: demisto_sdk/commands/common/errors.py [hunk 8]
```diff
@@ -1295,6 +1302,11 @@ def readme_equal_description_error():
         return 'README.md content is equal to pack description. ' \
                'Please remove the duplicate description from README.md file.'
 
+    @staticmethod
+    @error_code_decorator
+    def readme_contains_demisto_word(line_num):
+        return f'Found the word \'Demisto\' in the readme content in line {line_num}.'
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1308]:
```suggestion
        return f'Found the word \'Demisto\' in the readme content in lines: {line_num}.'
```

File: demisto_sdk/commands/common/errors.py [hunk 9]
```diff
@@ -1020,6 +1022,11 @@ def invalid_description_name():
         return "The description's file name is invalid - " \
                "make sure the name looks like the following: <integration_name>_description.md"
 
+    @staticmethod
+    @error_code_decorator
+    def description_contains_demisto_word(line_num):
+        return f'Found the word \'Demisto\' in the description content in line {line_num}.'
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:1028]:
```suggestion
        return f'Found the word \'Demisto\' in the description content in lines: {line_num}.'
```

**Review by roysagi — APPROVED** [review]
Good - just make sure to give a single error and not multiple ones.

File: demisto_sdk/commands/common/hook_validations/readme.py (hunk 7 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/hook_validations/readme.py:372]: done.

File: demisto_sdk/commands/common/hook_validations/description.py (hunk 6 above)
Comment (yaakovpraisler) [reply demisto_sdk/commands/common/hook_validations/description.py:176]: done.

**Pull request merged**